app/src/main/java/com/hover/stax/bounties/BountyChannelsAdapter.java

```diff
@@ -36,28 +34,18 @@ public CardedBountyListViewHolder onCreateViewHolder(@NonNull ViewGroup parent,
 
 	@Override
 	public void onBindViewHolder(@NonNull CardedBountyListViewHolder holder, int position) {
-		Channel c = channelList.get(position);
-		holder.staxCardView.setTitle(c.getUssdName());
-		List<Bounty> channelBounties = filterBounties(c.id);
```

Comment:
This causes nested loops, as onBindViewHolder itself is somewhat a list. Searching outside here would be better